### ISSUE TITLE
"excludeDevDependencies" option does not work without npm's package-lock.json

### ISSUE THREAD

**Issue by saibotsivad** [serverless/serverless#7563]
# Bug Report

## Description

1. What did you do?

- Used Yarn for managing dependencies, did not create a `package-lock.json` file (there's a `package.json` and a `yarn.lock` file instead).
- Install a dev dependency, e.g. `yarn add --dev serverless`.
- Set the flag `excludeDevDependencies: true` inside the `serverless.yml` file.
- Ran the command `export SLS_DEBUG=* && serverless package`

1. What happened?

- The command finished successfully.
- I see this in the logs: `Serverless: Excluding development dependencies...`
- The `zip` package file is created.
- When I unpack the `zip` file, I find the dev dependencies inside the `node_modules` folder.

1. What should've happened?

The dev dependencies are not included inside the zipped package file.

1. What's the content of your `serverless.yml` file?

```yaml
service: test
package:
  excludeDevDependencies: true
provider:
  name: aws
  runtime: nodejs12.x
  stage: dev
  region: us-east-1
functions:
  test:
    events:
      - http: GET /
```

Similar issues:

- #6090

Other notes:

I see in the `excludeNodeDevDependencies` function [this line](https://github.com/serverless/serverless/blob/8518000d4fbf3a6cf0a6e2f81bd6421e017a1b5f/lib/plugins/package/lib/zipService.js#L193):

```js
          return childProcess
            .execAsync(
              `npm ls --${env}=true --parseable=true --long=false --silent >> ${depFile}`,
              { cwd: dirWithPackageJson }
            )
```

I haven't tracked down the exact reason, but if you install dependencies with Yarn, the command `npm ls` fails.

**Comment by josephwarrick** [comment]
I ran into this issue; dev dependencies were included in the zip package when using yarn for dependency management.

In my case, my project had a dependency which depended on `npm`, so there was an older version of `npm` (6.5.0) in my node_modules directory.

For that old version of `npm`, `npm ls` was throwing an error for my version of node (12.18.3).

When serverless ran `npm ls --${env}=true --parseable=true --long=false --silent >> ${depFile}`, it was using the older version of `npm` from node_modules instead of the newer global `npm`, so the error got thrown (but swallowed and ignored), and the dev dependencies were not excluded.

To fix the problem, I upgraded the dependency, which in turn upgraded its dependent version of `npm`, and then everything worked as expected.
